src/charts/line/reusable-line-chart/reusable-line-chart.js

```diff
@@ -0,0 +1,669 @@
+import 'd3-transition';
+import {transition} from 'd3-transition';
+import {scaleLinear, scaleTime} from 'd3-scale';
+import {axisBottom, axisLeft} from 'd3-axis';
+import {easeLinear} from 'd3-ease';
+import {max, min} from 'd3-array';
+import {select} from 'd3-selection';
+import {format} from 'd3-format';
+import {line} from 'd3-shape';
+import d3Tip from "d3-tip";
+
+const margin = {top: 50, right: 50, bottom: 80, left: 60};
+
+Object.defineProperty(Array.prototype, 'flat', {
```

Comment:
Is this a polyfill? Could you feature test for it rather than blanket override it?

Reply:
And and add a comment indicating that's what's happening here

Reply:
You're right, this isn't actually needed, so removed these lines